### PR TITLE
[TECHDEBT] Add TestUtilityCodec in the Shared Module (issue-#231)

### LINKED ISSUE #231: [TECHDEBT] Add TestUtilityCodec in the Shared Module
## Objective

- Add unit test for UtilityCodec()

## Origin Document

TestUtilityCodec is missing in the `shared module` due to the need for a proto testing structure.

## Goals

- Increase testing code coverage

## Deliverable

- Code complete implementation

## General issue deliverables

- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

---

**Creator**: @andrewnguyen22

### PR DESCRIPTION
**Pull request by andrewnguyen22**
## Description

TestUtilityCodec is missing in the `shared module` due to the need for a proto testing structure.

- [x] Add unit test for `UtilityCodec()`
- [x]  Added `TestProtoStructure` for testing

## Issue

Fixes issue-#231

## Type of change

Please mark the options that are relevant.

- [ ] New feature, functionality or library
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [x] Other: Testing

## Testing
- [x] `make test_all`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [ ] If applicable, I have made corresponding changes to related local or global README
- [ ] If applicable, I have added new diagrams using [mermaid.js](https://mermaid-js.github.io)
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: shared/codec/proto/codec_test.proto [hunk 1]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+package utility;
+
+option go_package = "github.com/pokt-network/pocket/shared/codec";
+
+message TestProtoStructure {
```
Comment (Olshansk) [inline shared/codec/proto/codec_test.proto:6]: I don't fully understand the need/purpose of this test structure, which likely means it needs some additional documentation.

Can you explain what it's for? We can probably move that explanation in comments or a README for future readers too.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: shared/codec/proto/codec_test.proto (hunk 1 above)
Comment (andrewnguyen22) [reply shared/codec/proto/codec_test.proto:6]: Do you understand the basis of this issue? 

I can be more explicit in code if that's what you're asking

File: shared/codec/codec_test.go [hunk 2]
```diff
@@ -1,9 +1,63 @@
...
+
+func requireTestProtoStructureEqual(t *testing.T, tpsOne, tpsTwo *TestProtoStructure) {
+	require.Equal(t, tpsOne.Field1, tpsTwo.Field1)
+	require.Equal(t, tpsOne.Field2, tpsTwo.Field2)
+	require.Equal(t, tpsOne.Field3, tpsTwo.Field3)
+}
+
+func requireTestProtoStructureNotEqual(t *testing.T, tpsOne, tpsTwo *TestProtoStructure) {
+	allFieldsEqual := tpsOne.Field1 == tpsTwo.Field1 && tpsOne.Field2 == tpsTwo.Field2 && tpsOne.Field3 == tpsTwo.Field3
+	require.False(t, allFieldsEqual)
```
Comment (Olshansk) [inline shared/codec/codec_test.go:62]:
```suggestion
        require.NotEqual(t, tpsOne.Field1, tpsTwo.Field1)
        require.NotEqual(t, tpsOne.Field2, tpsTwo.Field2)
        require.NotEqual(t, tpsOne.Field3, tpsTwo.Field3)
```

File: shared/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.1] - 2022-09-24
+- Add unit test for `UtilityCodec()`
```
Comment (Olshansk) [inline shared/CHANGELOG.md:11]: I understand why you called in `UtilityCodec` and can see a world where we use different codecs for different modules (though it gets messy), but maybe we can rename it to `SharedCodec`? 

I think that's what through my understanding off originally.

File: shared/codec/proto/codec_test.proto [hunk 4]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+package utility;
```
Comment (Olshansk) [inline shared/codec/proto/codec_test.proto:2]: Maybe one of these instead?
s/utility/shared?
s/utility/codec?
s/utility/proto?

File: shared/codec/codec_test.go [hunk 5]
```diff
@@ -1,9 +1,63 @@
...
+	require.NoError(t, err)
+
+	tpsTwoProtoBytes, err := codec.Marshal(tpsTwo)
+	require.NoError(t, err)
+	require.NotEqual(t, tpsOneProtoBytes, tpsTwoProtoBytes)
+
+	// test unmarshalling
+	tpsOneUnmarshalled := &TestProtoStructure{}
+	require.NoError(t, codec.Unmarshal(tpsOneProtoBytes, tpsOneUnmarshalled))
+	require.NoError(t, err)
```
Comment (Olshansk) [inline shared/codec/codec_test.go:43]:
```suggestion
	require.NoError(t, err)
	require.NoError(t, codec.Unmarshal(tpsOneProtoBytes, tpsOneUnmarshalled))
```

File: shared/codec/codec_test.go [hunk 6]
```diff
@@ -1,9 +1,63 @@
...
+
+	tpsTwoUnmarshalled := &TestProtoStructure{}
+	require.NoError(t, codec.Unmarshal(tpsTwoProtoBytes, tpsTwoUnmarshalled))
+	require.NoError(t, err)
```
Comment (Olshansk) [inline shared/codec/codec_test.go:47]:
```suggestion
	require.NoError(t, err)
	require.NoError(t, codec.Unmarshal(tpsTwoProtoBytes, tpsTwoUnmarshalled))
```

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Review by Olshansk — CHANGES_REQUESTED** [review]
Need a merge with master but LGTM otherwise

**Review by Olshansk — APPROVED** [review]

**Pull request merged**